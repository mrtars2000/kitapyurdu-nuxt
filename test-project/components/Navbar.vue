<template>
  <v-sheet>
    <v-app-bar
      color="black"
      dense
    >
      <v-toolbar-items class="white--text">
        <v-btn to="Login" style="background-color: orange; box-shadow: none;">
          Giriş Yap
        </v-btn>
      </v-toolbar-items>

      <v-toolbar-items>
        <v-btn to="Register" style="background-color:black;border-color: transparent; box-shadow:none;" class="mr-5 white--text">
          Üye ol
        </v-btn>
      </v-toolbar-items>

      <v-spacer />

      <v-menu
        offset-y
      >
        <template #activator="{ on, attrs }">
          <v-btn
            icon
            v-bind="attrs"
            class="white--text"
            style="margin-right: 75px;"
            v-on="on"
          >
            Listelerim
          </v-btn>
        </template>

        <v-list>
          <v-list-item
            v-for="(list,index) in lists"
            :key="index"
            @click="() => {}"
          >
            <v-list-item-title>{{ list.title }}</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
      <v-menu
        offset-y
      >
        <template #activator="{ on, attrs }">
          <v-btn
            icon
            v-bind="attrs"
            style="margin-right: 75px;"
            class="white--text"
            v-on="on"
          >
            Sipariş Takibi
          </v-btn>
        </template>

        <v-list>
          <v-list-item
            v-for="(siparisTakibi,index) in siparisTakibis"
            :key="index"
            @click="() => {}"
          >
            <v-list-item-title>{{ siparisTakibi.title }}</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
      <v-menu
        offset-y
      >
        <template #activator="{ on, attrs }">
          <v-btn
            icon
            v-bind="attrs"
            style="margin-right: 50px;"
            class="white--text"
            v-on="on"
          >
            Türkçe
          </v-btn>
        </template>

        <v-list>
          <v-list-item
            v-for="(language,index) in languages"
            :key="index"
            @click="() => {}"
          >
            <v-list-item-title>{{ language.title }}</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
      <v-menu
        offset-y
      >
        <template #activator="{ on, attrs }">
          <v-btn
            icon
            v-bind="attrs"
            style="margin-right: 50px;"
            class="white--text"
            v-on="on"
          >
            TL
          </v-btn>
        </template>

        <v-list>
          <v-list-item
            v-for="(money,index) in moneys"
            :key="index"
            @click="() => {}"
          >
            <v-list-item-title>{{ money.title }}</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>

      <v-menu
        offset-y
      >
        <template #activator="{ on, attrs }">
          <v-btn
            icon
            v-bind="attrs"
            class="white--text"
            style="margin-right: 25px;"
            v-on="on"
          >
            Yardım
          </v-btn>
        </template>

        <v-list>
          <v-list-item
            v-for="(help,index) in helps"
            :key="index"
            :menu-props="{bottom:true,offsetY:true}"
            @click="() => {}"
          >
            <v-list-item-title>{{ help.title }}</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
    </v-app-bar>

    <div>
      <v-app-bar
        color="white"
      >
        <v-toolbar-title style="font-weight: bold;" class="green--text" to="/">
          <NuxtLink style="font-weight: bold; text-decoration:none" class="green--text" to="/">
            Kitapyurdu.com
          </NuxtLink>
        </v-toolbar-title>

        <v-spacer />

        <v-text-field
          style="margin-top: 25px; "
          placeholder="kitap adı,yazar veya yayın evi"
          dense
          solo
        />

        <v-btn style="background-color: orange;" to="ProductFiltering">
          <v-icon>mdi-magnify </v-icon>
        </v-btn>

        <v-spacer />
        <div class="text-center">
          <v-menu offset-y>
            <template #activator="{ on, attrs }">
              <v-btn
                dark
                v-bind="attrs"
                style="margin-right: 25px; background-color: transparent; box-shadow: none;"
                v-on="on"
              >
                <p style="color: black;" class="my-2 mx-1">
                  Sepetim
                </p>

                <v-icon style="color: orange;">
                  mdi-cart
                </v-icon>
              </v-btn>
            </template>
            <v-card style="width: 400px;">
              <div>
                <v-row
                  v-for="(item, i) in baskets"
                  :key="i"
                >
                  <v-col>
                    <v-card>
                      <div class="d-flex flex-no-wrap justify-space-between">
                        <v-avatar
                          class="ma-3"
                          size="60"
                          tile
                        >
                          <v-img :src="item.src" contain />
                        </v-avatar>
                        <div>
                          <v-card-title
                            class="text-h5"
                            v-text="item.title"
                          />

                          <v-card-subtitle v-text="item.id" />
                        </div>
                      </div>
                      <v-divider class="mx-4" />
                      <v-card-text>
                        <v-chip-group
                          v-model="selection"
                          column
                        >
                          <v-chip class="orange">
                            Sepete git
                          </v-chip>
                          <v-spacer />
                          <v-chip class="red">
                            Satın al
                          </v-chip>
                        </v-chip-group>
                      </v-card-text>
                    </v-card>
                  </v-col>
                </v-row>
              </div>
            </v-card>
          </v-menu>
        </div>
      </v-app-bar>
    </div>
  </v-sheet>
</template>

<script>

export default {
  data () {
    return {
      helps: [
        { title: 'Yardım ve İşlem Rehberi' },
        { title: 'Sıkça Sorulan Sorular' },
        { title: 'Video Yardım' },
        { title: 'Önerim Var' },
        { title: 'İletişim' }
      ],
      lists: [
        { title: 'Favorilerim' },
        { title: 'Alışveriş Listem' },
        { title: 'Okuma Listelerim' },
        { title: 'Uyarım Listem' },
        { title: 'Satın Aldıklarım' }

      ],
      siparisTakibis: [
        { title: 'Şiparişlerim' },
        { title: 'Müşteri Hizmetler' },
        { title: 'Havale/EFT' }
      ],
      languages: [
        { title: 'Türkçe' },
        { title: 'English' },
        { title: 'Deutsch' }
      ],
      moneys: [
        { title: 'TL' },
        { title: 'EUR' },
        { title: 'GBP' },
        { title: 'USD' }
      ]
    }
  }
}
</script>

<style>
  .a{
    opacity: 1;
    color: black;
  }

</style>
