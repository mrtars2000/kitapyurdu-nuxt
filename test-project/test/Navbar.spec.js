import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import Navbar from '@/components/Navbar'
Vue.use(Vuetify)
Vue.use(Vuex)

const factory = () => {
  return shallowMount(Navbar, {})
}

describe('Navbar.vue', () => {
  // eslint-disable-next-line no-unused-vars
  let vuetify
  let store

  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    vuetify = new Vuetify()

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    store = new Vuex.Store({
      data () {
        return {
          helps: [
            { title: 'Yardım ve İşlem Rehberi' },
            { title: 'Sıkça Sorulan Sorular' },
            { title: 'Video Yardım' },
            { title: 'Önerim Var' },
            { title: 'İletişim' }
          ],
          lists: [
            { title: 'Favorilerim' },
            { title: 'Alışveriş Listem' },
            { title: 'Okuma Listelerim' },
            { title: 'Uyarım Listem' },
            { title: 'Satın Aldıklarım' }

          ],
          siparisTakibis: [
            { title: 'Şiparişlerim' },
            { title: 'Müşteri Hizmetler' },
            { title: 'Havale/EFT' }
          ],
          languages: [
            { title: 'Türkçe' },
            { title: 'English' },
            { title: 'Deutsch' }
          ],
          moneys: [
            { title: 'TL' },
            { title: 'EUR' },
            { title: 'GBP' },
            { title: 'USD' }
          ]
        }
      }
    })
  })

  const wrapper = factory()

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('Snapshot property ', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
