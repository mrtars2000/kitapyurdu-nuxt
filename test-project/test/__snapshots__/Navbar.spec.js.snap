// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue Snapshot property  1`] = `
"<v-sheet-stub tag=\\"div\\">
  <v-app-bar-stub color=\\"black\\" tag=\\"header\\" dense=\\"true\\" extensionheight=\\"48\\" src=\\"\\" value=\\"true\\">
    <v-toolbar-items-stub tag=\\"div\\" class=\\"white--text\\">
      <v-btn-stub tag=\\"button\\" activeclass=\\"\\" to=\\"Login\\" type=\\"button\\" style=\\"background-color: orange; box-shadow: none;\\">
        Giriş Yap
      </v-btn-stub>
    </v-toolbar-items-stub>
    <v-toolbar-items-stub tag=\\"div\\">
      <v-btn-stub tag=\\"button\\" activeclass=\\"\\" to=\\"Register\\" type=\\"button\\" class=\\"mr-5 white--text\\" style=\\"background-color: black; border-color: transparent; box-shadow: none;\\">
        Üye ol
      </v-btn-stub>
    </v-toolbar-items-stub>
    <v-spacer-stub tag=\\"div\\"></v-spacer-stub>
    <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" closeoncontentclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"v-menu-transition\\">
      <v-list-stub tag=\\"div\\">
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Favorilerim</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Alışveriş Listem</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Okuma Listelerim</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Uyarım Listem</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Satın Aldıklarım</v-list-item-title-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-menu-stub>
    <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" closeoncontentclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"v-menu-transition\\">
      <v-list-stub tag=\\"div\\">
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Şiparişlerim</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Müşteri Hizmetler</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Havale/EFT</v-list-item-title-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-menu-stub>
    <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" closeoncontentclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"v-menu-transition\\">
      <v-list-stub tag=\\"div\\">
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Türkçe</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">English</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">Deutsch</v-list-item-title-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-menu-stub>
    <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" closeoncontentclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"v-menu-transition\\">
      <v-list-stub tag=\\"div\\">
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">TL</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">EUR</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">GBP</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
          <v-list-item-title-stub tag=\\"div\\">USD</v-list-item-title-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-menu-stub>
    <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" closeoncontentclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"v-menu-transition\\">
      <v-list-stub tag=\\"div\\">
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\" menu-props=\\"[object Object]\\">
          <v-list-item-title-stub tag=\\"div\\">Yardım ve İşlem Rehberi</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\" menu-props=\\"[object Object]\\">
          <v-list-item-title-stub tag=\\"div\\">Sıkça Sorulan Sorular</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\" menu-props=\\"[object Object]\\">
          <v-list-item-title-stub tag=\\"div\\">Video Yardım</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\" menu-props=\\"[object Object]\\">
          <v-list-item-title-stub tag=\\"div\\">Önerim Var</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass=\\"\\" tag=\\"div\\" menu-props=\\"[object Object]\\">
          <v-list-item-title-stub tag=\\"div\\">İletişim</v-list-item-title-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-menu-stub>
  </v-app-bar-stub>
  <div>
    <v-app-bar-stub color=\\"white\\" tag=\\"header\\" extensionheight=\\"48\\" src=\\"\\" value=\\"true\\">
      <v-toolbar-title-stub tag=\\"div\\" to=\\"/\\" class=\\"green--text\\" style=\\"font-weight: bold;\\">
        <nuxtlink to=\\"/\\" class=\\"green--text\\" style=\\"font-weight: bold; text-decoration: none;\\">
          Kitapyurdu.com
        </nuxtlink>
      </v-toolbar-title-stub>
      <v-spacer-stub tag=\\"div\\"></v-spacer-stub>
      <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" dense=\\"true\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" placeholder=\\"kitap adı,yazar veya yayın evi\\" solo=\\"true\\" type=\\"text\\" style=\\"margin-top: 25px;\\"></v-text-field-stub>
      <v-btn-stub tag=\\"button\\" activeclass=\\"\\" to=\\"ProductFiltering\\" type=\\"button\\" style=\\"background-color: orange;\\">
        <v-icon-stub>mdi-magnify </v-icon-stub>
      </v-btn-stub>
      <v-spacer-stub tag=\\"div\\"></v-spacer-stub>
      <div class=\\"text-center\\">
        <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" closeoncontentclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"v-menu-transition\\">
          <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" style=\\"width: 400px;\\">
            <div></div>
          </v-card-stub>
        </v-menu-stub>
      </div>
    </v-app-bar-stub>
  </div>
</v-sheet-stub>"
`;
